src/js/view/view-model.ts

```diff
@@ -63,31 +72,34 @@ class PlayerViewModel extends SimpleModel {
         }
     }
 
-    get(attr) {
+    get(attr: keyof PlayerModelAttributes | keyof MediaModelAttributes): any {
         const mediaModel = this._mediaModel;
         if (mediaModel && attr in mediaModel.attributes) {
-            return mediaModel.get(attr);
+            return mediaModel.get(attr as keyof MediaModelAttributes);
         }
-        return this._model.get(attr);
+        return this._model.get(attr as keyof PlayerModelAttributes);
     }
 
-    set(attr, val) {
+    set(attr: keyof PlayerModelAttributes, val: any): void {
         return this._model.set(attr, val);
     }
 
-    getVideo() {
+    getVideo(): DefaultProvider | null {
```

Comment:
could use a union type here as well for all providers?

Reply:
Base class is good. Most providers are defined outside of this project.